src/blocks/weather.rs

```diff
@@ -153,7 +153,7 @@ fn default_interval() -> Seconds {
 trait WeatherProvider {
     async fn get_weather(
         &self,
-        autolocated_location: Option<Coordinates>,
+        autolocated_location: &Option<Coordinates>,
```

Comment:
Usually it is better to use `Option<&T>` instead of `&Option<T>`. Doesn't matter as much in this case though.